lasagne/tests/layers/test_conv.py

```diff
@@ -566,10 +593,16 @@ def test_with_nones(self, DummyInputLayer, input, kernel, output, kwargs):
                 filter_size=kernel.shape[2:],
                 W=kernel.transpose(1, 0, 2, 3),
                 **kwargs)
-        assert layer.output_shape == (None, output.shape[1], None, None)
+        if 'output_size' not in kwargs or isinstance(kwargs['output_size'],
+                                                     T.Variable):
+            assert layer.output_shape == (None, output.shape[1], None, None)
         actual = layer.get_output_for(input).eval()
         assert actual.shape == output.shape
         assert np.allclose(actual, output)
+        # Check get_output_shape_for for non symbolic output
+        if 'output_size' in kwargs and not isinstance(kwargs['output_size'],
+                                                      T.Variable):
+            assert layer.get_output_shape_for(input.shape) == output.shape
```

Comment:
That's an interesting test as well, but I had though of:
```python
assert layer.output_shape = (None, output.shape[1]) + kwargs['output_size']
```
I.e., the layer should report the output size even when it doesn't know most of the input size. Can you add this? (Not replace your current line, just add this one. Both seem useful.)